docs/en/contributing-to-appium/appium-packages.md

```diff
@@ -0,0 +1,255 @@
+- builds AppiumBootstrap.jar that contains logic to execute the commands
+- counterpart to appium-uiauto
+- once started it creates a web socket connection to the device
+  - application provides start/shutdown/sendCommand interface
+- command flow is like:
+  - Selenium command -> appium-adb -> appium-android-bootstrap -> Java code using the Android UI Automator framework
+
+## appium-uiautomator
+- starts and shutdowns uiautomator server given by appium-android-bootstrap jar build
+- command flow is like
+  - appium-android-bootstrap:start -> appium-uiautomator:start -> appium-adb:install bootstrap
+
+## appium-selendroid-driver
+- similar to appium-android-driver it can run as standalone server
+- downloads and installs Selendroid using appium-selendroid-installer
+- contains several Selendroid specific logic to ensure a seamless integration
+- contains a more specific set of capability constraints
+- uses jsonwp-proxy to talk to the server
+- used appium-adb to enable commands not implemented in Selendroid
+
+## appium-selendroid-installer
+- contains and exports a setup logic to
+  - download Selendroid
+  - determine AndroidManifest location
+  - determine Server APK location
+  - extracting both files
+  - copying and cleaning files
+
+## appium-android-ime
+- allows to send and receive unicode characters from/to the Android device
+- encodes text into UTF-7 sends it to the device and recodes it as Unicode
+- used by appium-android-driver and appium-selendroid-driver
+
+## appium-doctor
+- diagnoses, reports and fixes common Node, iOS and Android configuration issues before starting Appium
+- exposes cli command “appium-doctor"
+- it checks for
+  - Android:
+    - android sdk exists and configured properly
+    - env variables and path check
+  - iOS:
+    - xcode is installed (with command line tools)
+    - dev tools security check
+    - auth check
+    - node binary check
+
+## appium-gulp-plugins
+- dev package with custom plugins used accross appium modules (for Appium development only)
+- contains task for
+  - e2e and unit tests (with coverage reporting)
+  - transpiling ES2016 into ES5
+  - static code analysis (jshint)
+  - watch task for dev
+
+## appium-remote-debugger
+- RPC client to connect Appium to iOS webviews
+- can connect to WebKit devtools
+- for iOS only
+- has two rpc client classes
+  - remote-debugger-rpc-client: uses tcp6 that connects to localhost:27753
+  - webkit-rpc-client: uses WebSockets to connect to ws://localhost:27753/devtools/page/${pageId}
+
+## node-teen_process
+- helper module that exposes:
+  - exec: ES7 (async/await) implementation of exec that uses spawn under the hood
+  - SubProcess: cuts down boilerplate when using spawn (especially when using in an async/await context)
+
+## appium-logger
+- basic logger defaulting to npmlog with special consideration for running tests
+- exposes getLogger function that gets used by almost all Appium packages
```

Comment:
Also defers to already-running logger if there is one, so everything bubbles up.